nncf/common/sparsity/schedulers.py

```diff
@@ -74,20 +74,21 @@ def _update_sparsity_level(self) -> None:
         Calculates the current sparsity level and updates the internal
         state of the `controller`.
         """
-        self._current_level = self._calculate_sparsity_level()
         if self.current_epoch >= self.freeze_epoch:
             self._controller.freeze()
-        self._controller.set_sparsity_level(self._current_level)
+        self._controller.set_sparsity_level(self._calculate_sparsity_level())
 
     @property
-    def current_sparsity_level(self) -> float:
+    def current_sparsity_level(self) -> Union[float, None]:
```

Comment:
`Optional[float]`. Should also document in the docstring specifically the situations when `None` may be returned.

Reply:
Thanks for the comment. Fixed!